File: backend/list_attempts.js
const admin = require('firebase-admin');
const serviceAccount = require('./serviceAccountKey.json');

admin.initializeApp({
  credential: admin.credential.cert(serviceAccount)
});

const db = admin.firestore();

async function run() {
  const snapshot = await db.collection('testAttempts').get();
  console.log("Total testAttempts:", snapshot.size);
  snapshot.forEach(doc => {
    console.log(doc.id, doc.data());
  });
}

run();

File: backend/list_attempts.py
import firebase_admin
from firebase_admin import credentials, firestore

try:
    cred = credentials.Certificate("serviceAccountKey.json")
    firebase_admin.initialize_app(cred)
except:
    pass # Might already be initialized in some environments, or we just rely on default credentials if running in a certain way

try:
    from app.firebase import db
except:
    db = firestore.client()

attempts = db.collection('testAttempts').get()
print("Total testAttempts:", len(attempts))
for attempt in attempts:
    print(attempt.id, attempt.to_dict())
